test/sockets.test.js

```diff
@@ -1,63 +1,186 @@
 import 'regenerator-runtime/runtime';
 
 import Socket from '../src/sockets';
-import Logger from '../src/logger';
+import { Logger, SOCKET_PING } from 'syft-helpers.js';
 
 import { WebSocket, Server } from 'mock-socket';
 
 global.WebSocket = WebSocket;
 
 const url = 'ws://localhost:8080/';
 
-const onMessage = event => console.log('NEW MESSAGE', event);
-const onOpen = event => console.log('NEW OPEN', event);
-const onClose = event => console.log('NEW CLOSE', event);
-const onError = event => console.log('NEW ERROR', event);
+/**
+ * Creates promise from the first event call
+ * @param emitter Object that emits the event
+ * @param event Event name
+ * @returns {Promise}
+ */
+function makeEventPromise(emitter, event) {
+  let resolver;
+  let promise = new Promise(resolve => resolver = resolve);
+  emitter.on(event, data => resolver(data));
+  return promise;
+}
 
 describe('Sockets', () => {
+  // common test objects
+  let mockServer;
+
   jest.spyOn(console, 'log');
 
+  beforeEach(() => {
+    mockServer = new Server(url);
+    mockServer.connected = makeEventPromise(mockServer, 'connection');
+  });
+
   afterEach(() => {
+    mockServer.close();
     jest.clearAllMocks();
   });
 
-  test('can communicate with a socket server', async () => {
-    const message = { message: 'This is a test' };
+  test('keep-alive messages', async() => {
+    const keepAliveTimeout = 300;
+    const expectedMessagesCount = 3;
 
-    const mockServer = new Server(url);
     const mySocket = new Socket({
       url,
       logger: new Logger(true),
-      onMessage: event => onMessage(event),
-      onOpen: event => onOpen(event),
-      onClose: event => onClose(event),
-      onError: event => onError(event)
+      keepAliveTimeout
     });
+    let serverSocket = await mockServer.connected;
+    // collect messages
+    let messages = [];
+    serverSocket.on('message', message => messages.push(JSON.parse(message)));
+    // if we wait long enough, we should see `expectedMessagesCount` keep-alive messages
+    await new Promise(done => setTimeout(done,
+      keepAliveTimeout * expectedMessagesCount + keepAliveTimeout / 2));
+    // one message is sent right after establishing the connection hence +1 
+    // (do we really need that?)
+    expect(messages).toHaveLength(expectedMessagesCount + 1);
+    let expectedTypes  =[];
+    for (let i = 0; i < expectedMessagesCount + 1; i++) {
+      expectedTypes.push(SOCKET_PING);
+    }
+    expect(messages.map(message => message["type"])).toEqual(expectedTypes);
+  });
 
-    expect(console.log.mock.calls.length).toBe(0);
+  test('onopen event works', async () => {
+    const onOpen = jest.fn();
+    const mySocket = new Socket({
+      url,
+      logger: new Logger(true),
+      onOpen
+    });
+    await mockServer.connected;
```

Comment:
Give a little spacing here.  I like to keep my code spaced out as much as possible, allowing for variables to be declared at the top and then have logic sections of code separated by two newlines (one blank line between logical statements).

Reply:
Done